Spigot-Server-Patches/0673-Add-PlayerGlassBottleFillEvent.patch

```diff
@@ -0,0 +1,123 @@
+@@ -1,5 +1,7 @@
+ package net.minecraft.server;
+ 
++import io.papermc.paper.event.player.PlayerGlassBottleFillEvent; // Paper
++
+ import java.util.List;
+ 
+ public class ItemGlassBottle extends Item {
+@@ -20,7 +22,10 @@ public class ItemGlassBottle extends Item {
+ 
+             entityareaeffectcloud.setRadius(entityareaeffectcloud.getRadius() - 0.5F);
+             world.playSound((EntityHuman) null, entityhuman.locX(), entityhuman.locY(), entityhuman.locZ(), SoundEffects.ITEM_BOTTLE_FILL_DRAGONBREATH, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+-            return InteractionResultWrapper.a(this.a(itemstack, entityhuman, new ItemStack(Items.DRAGON_BREATH)), world.s_());
++            // Paper start - PlayerGlassBottleFillEvent
++            //return InteractionResultWrapper.a(this.a(itemstack, entityhuman, new ItemStack(Items.DRAGON_BREATH)), world.s_());
++            return handleInteractedItemStack(world, entityhuman, itemstack, new ItemStack(Items.DRAGON_BREATH));
++            // Paper end - PlayerGlassBottleFillEvent
+         } else {
+             MovingObjectPositionBlock movingobjectpositionblock = a(world, entityhuman, RayTrace.FluidCollisionOption.SOURCE_ONLY);
+ 
+@@ -36,7 +41,10 @@ public class ItemGlassBottle extends Item {
+ 
+                     if (world.getFluid(blockposition).a((Tag) TagsFluid.WATER)) {
+                         world.playSound(entityhuman, entityhuman.locX(), entityhuman.locY(), entityhuman.locZ(), SoundEffects.ITEM_BOTTLE_FILL, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+-                        return InteractionResultWrapper.a(this.a(itemstack, entityhuman, PotionUtil.a(new ItemStack(Items.POTION), Potions.WATER)), world.s_());
++                        // Paper start - PlayerGlassBottleFillEvent
++                        //return InteractionResultWrapper.a(this.a(itemstack, entityhuman, PotionUtil.a(new ItemStack(Items.POTION), Potions.WATER)), world.s_());
++                        return handleInteractedItemStack(world, entityhuman, itemstack, PotionUtil.a(new ItemStack(Items.POTION), Potions.WATER));
++                        // Paper end - PlayerGlassBottleFillEvent
+                     }
+                 }
+ 
+@@ -45,6 +53,18 @@ public class ItemGlassBottle extends Item {
+         }
+     }
+ 
++    // Paper start - PlayerGlassBottleFillEvent
++    protected InteractionResultWrapper<ItemStack> handleInteractedItemStack(World world, EntityHuman entityhuman, ItemStack itemstack, ItemStack newitemstack) {
++        PlayerGlassBottleFillEvent event = new PlayerGlassBottleFillEvent((org.bukkit.entity.Player) entityhuman.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(newitemstack));
++        if (event.callEvent()) {
++            return InteractionResultWrapper.a(this.a(itemstack, entityhuman, org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getNewItem())), world.s_());
```

Comment:
add a note here that this line was copied from above. I think it doesnt need obf helpers then, as its easy to find the new names by just copying the line again

Reply:
I've added. Is it right?